src/server/_common.py

```diff
@@ -18,12 +20,37 @@ def _get_db() -> Connection:
         g.db = conn
     return g.db
 
+def log_and_raise_exception(message):
+    app.logger.error(message)
+    raise ValidationFailedException(message)
 
 """
 access to the SQL Alchemy connection for this request
 """
 db: Connection = cast(Connection, LocalProxy(_get_db))
 
+@event.listens_for(engine, "before_cursor_execute")
+def before_cursor_execute(conn, cursor, statement, parameters, context, executemany):
+    t = time.time()
+    # TODO: probbly just use one of these two:
+    context._query_start_time = t
+    conn.info.setdefault('query_start_time', []).append(t)
+
+
+@event.listens_for(engine, "after_cursor_execute")
+def after_cursor_execute(conn, cursor, statement, parameters, context, executemany):
+    # this timing info may be suspect, at least in terms of dmbs cpu time...
+    # it is likely that it includes that time as well as any overhead that
+    # come from throttling or flow control on the streamed data, as well as
+    # any row transform/processing time
+    t = time.time()
+    total_cntx = t - context._query_start_time
+    total_cnxn = t - conn.info['query_start_time'].pop(-1)
+    # Convert to milliseconds
+    total_cntx = total_cntx * 1000
+    total_cnxn = total_cnxn * 1000
+    app.logger.info("SQL execute() elapsed time: %s", str(dict(
+        statement=statement, total_cntx=total_cntx, total_cnxn=total_cnxn)))
```

Comment:
gross. does flask logger support structured logging or are we stuck parsing the dict out of the string downstream?

Reply:
I found https://www.structlog.org/en/22.1.0/examples.html#flask-and-thread-local-data which suggests we can adapt this for structlog

apologies for not thinking to point you at our existing structlog usage!
https://github.com/cmu-delphi/delphi-epidata/blob/dev/src/acquisition/covidcast/logger.py